### PR TITLE
Protect against WebPurify outage 

### PR DESCRIPTION
**Pull request by sureshc**
Set TCP connect and HTTP read timeout on WebPurify API calls and ensure that a Gatekeeper flag can be used to manually disable the calls. Together these changes ensure that performance and availability issues with this 3rd party WebService do not tie up all puma worker threads in our web application servers and cause an outage on our systems.

There are Gatekeeper checks [on some](https://github.com/code-dot-org/code-dot-org/blob/c2b29e9a75cc6ca56d00cf8fa85b1ea3957676cd/lib/cdo/share_filtering.rb#L58-L74), [but not all](https://github.com/code-dot-org/code-dot-org/blob/89f8b98fb6b7414d49bfc420c011016020454ca7/dashboard/app/controllers/profanity_controller.rb#L15), of the components that invoke this method.

[NewRelic metrics indicate](https://one.newrelic.com/launcher/nr1-core.explorer?pane=eyJuZXJkbGV0SWQiOiJhcG0tbmVyZGxldHMuZXh0ZXJuYWxzIiwiZW50aXR5SWQiOiJOVEF4TkRZemZFRlFUWHhCVUZCTVNVTkJWRWxQVG53ek56ZzNNelkwIiwic2VsZWN0ZWRTZXJpZXMiOiJiZmI1ZWY1ZjQ5NDg5Y2NjOTdiODhhOTM3NzFkMjcxNDhkMDU0NmUzIiwiZHJpbGxkb3duIjp7ImV4dGVybmFsLmhvc3QiOiJhcGkxLndlYnB1cmlmeS5jb20ifX0=&sidebars[0]=eyJuZXJkbGV0SWQiOiJucjEtY29yZS5hY3Rpb25zIiwiZW50aXR5SWQiOiJOVEF4TkRZemZFRlFUWHhCVUZCTVNVTkJWRWxQVG53ek56ZzNNelkwIiwic2VsZWN0ZWROZXJkbGV0Ijp7Im5lcmRsZXRJZCI6ImFwbS1uZXJkbGV0cy5leHRlcm5hbHMifX0=&platform[timeRange][duration]=604800000&platform[$isFallbackTimeRange]=false) that requests to this WebService rarely exceed 200ms, so 5 second TCP connect and 10 second HTTP read timeout should be sufficient

image

### Background

https://codedotorg.atlassian.net/browse/INF-396

## Links

- [spec]()
- [jira]()

## Testing story

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: lib/cdo/web_purify.rb [hunk 1]
```diff
@@ -27,7 +27,7 @@ def self.find_potential_profanities(text, language_codes = ['en'])
       "&text=#{URI.encode(text)}" \
       "&lang=#{language_codes}" \
       "&format=json"
-    result = JSON.parse(open(url).read)
+    result = JSON.parse(open(url, open_timeout: 5, read_timeout: 10).read)
```
Comment (sureshc) [inline lib/cdo/web_purify.rb]: Should we store these values in DCDO?

Comment (maddiedierker) [reply lib/cdo/web_purify.rb]: yeah, i think so. a 10-second read timeout is pretty safe, but if you do end up needing to tweak it, it's nice to have that option available immediately, especially for something we use a lot like Webpurify

Comment (sureshc) [reply lib/cdo/web_purify.rb]: That makes sense. I've added that. Let me know if I went to far with breaking this out into multiple, indented lines :)

**Review by maddiedierker — APPROVED** [review]
🎉  thank you for adding this!

**Pull request merged**